libmamba/src/util/string.cpp

```diff
@@ -447,6 +447,27 @@ namespace mamba::util
         return strip_if(input, [](Char c) { return !is_graphic(c); });
     }
 
+    void inplace_strip(std::string& input)
+    {
+        // Dedicated implementation for std::string to avoid copies
+        if (input.empty())
+        {
+            return;
+        }
+
+        const auto start = input.find_first_not_of(" \t\n\v\f\r");
+
+        if (start == std::string::npos)
+        {
+            input.clear();
+            return;
+        }
+
+        const auto end = input.find_last_not_of(" \t\n\v\f\r");
+
+        input = input.substr(start, end - start + 1);
```

Comment:
I think this is fine, but if we discover that this can create another string (not sure), then the alternative is to do 2 `.erase`, starting with the end.
Anyway that's a side-note, I think that's fine.

Reply:
Do you have an example? Would such a processing qualifies as YAGNI?

Reply:
I meant something like
```c++
input.erase(end, input.end());
input.erase(input.begin(), start);
```
(didnt check if it's correct)

> Would such a processing qualifies as YAGNI?

Yes, I dont think we need to worry about this, it's just a side note.

I was wondering if the implementation of `string x = x.substr(b,e);` is allowed to create a string. The standard spec says this is equivalent to `x.assign(x.substr(b, e));` and `assign` which is linear complexity in that case. I would expect all implementations just overwrite without new allocations, therefore this code should be fine. 
My comment is just a side note in case we realize later that it's not doing what we expect (and then the function is not working in-place as advertised).